### ISSUE TITLE
Reading the Exif in the load background thread

### ISSUE THREAD

**Issue by perracolabs** [bumptech/glide#3367]
**Glide Version**: 4.8.0

Since Glide doesn’t have Exif support for **raw (dng)** image files, we are currently using the **Exif Support Library** to read the correct orientation for such format and supply a custom transform class according to it.

The problem is that reading the Exif in the UI thread right before committing to Glide the task to load the image has a performance hit.

If Exif support for such format isn’t something in the roadmap, then, is there any callback in the load pipeline triggered in the **loading background thread** that we could use read the Exif orientation and supply the transform to be applied?

**Comment by sjudd** [comment]
On OMR 1 and higher we already use ExifInterface in the support library: https://github.com/bumptech/glide/blob/f7d860412f061e059aa84a42f2563a01ac8c303b/library/src/main/java/com/bumptech/glide/load/resource/bitmap/ExifInterfaceImageHeaderParser.java#L23

What version are you loading RAW images on that are incorrectly oriented? 

You could write a custom decoder to extract the orientation similar to Downsampler, but you probably don't want to. Otherwise there's no explicit plan for a callback to support this.

**Comment by perracolabs** [comment]
The problem is from Android N.
I had a look at the **ImageHeaderParser** source, and this implementation will only work for android O as it uses the native android.media.ExifInterface, instead of the **ExifInterface support library** which gives all the Exif latest capabilities to all android versions, including the RAW tags.

Does Glide use the support libraries or it relies only in the native implementations? 

These are the current ExifInterface support library versions depending on the type of support library type:

implementation 'com.android.support:exifinterface:28.0.0-rc02'
implementation 'androidx.exifinterface:exifinterface:1.0.0-rc02'

And the library introduction a while ago:

[ExifInterface support library](https://android-developers.googleblog.com/2016/12/introducing-the-exifinterface-support-library.html)

**Comment by perracolabs** [comment]
I agree, and it would make more sense to use the support library version instead of the framework one, as it gives the current and future bug fixes and new supported formats to all android versions. Fortunately, this is a single source file library, so size wise shouldn’t have much impact.
